tests/infer/test_abstract_infer.py

```diff
@@ -44,3 +46,30 @@ def nested():
         nested()
 
     assert len(tp.trace.nodes) == 0
+
+
+def test_information_criterion():
+    # milk dataset: https://github.com/rmcelreath/rethinking/blob/master/data/milk.csv
+    kcal = torch.tensor([0.49, 0.47, 0.56, 0.89, 0.92, 0.8, 0.46, 0.71, 0.68,
+                         0.97, 0.84, 0.62, 0.54, 0.49, 0.48, 0.55, 0.71])
+    kcal_mean = kcal.mean()
+    kcal_logstd = kcal.std().log()
+
+    def model():
+        mu = pyro.sample("mu", dist.Normal(kcal_mean, 1))
+        log_sigma = pyro.sample("log_sigma", dist.Normal(kcal_logstd, 1))
+        with pyro.plate("plate"):
+            pyro.sample("kcal", dist.Normal(mu, log_sigma.exp()), obs=kcal)
+
+    delta_guide = AutoLaplaceApproximation(model)
+
+    svi = SVI(model, delta_guide, optim.Adam({"lr": 0.05}), loss=Trace_ELBO(), num_samples=5000)
```

Comment:
@neerajprad I change `num_samples` to 5000 and delete the seed to have more stable result. I test this will different runs in CPU. Hope that it would be fine for GPU too.

Reply:
Don't worry about the GPU, @fehiepsi.